Jetsurvey/app/src/main/java/com/example/compose/jetsurvey/signinsignup/SignInScreen.kt

```diff
@@ -56,36 +57,58 @@ sealed class SignInEvent {
     object NavigateBack : SignInEvent()
 }
 
+@OptIn(ExperimentalMaterialApi::class)
 @Composable
 fun SignIn(onNavigationEvent: (SignInEvent) -> Unit) {
-    val showSnackbar = remember { mutableStateOf(false) }
-    Stack(modifier = Modifier.fillMaxSize()) {
-        SignInSignUpScreen(
-            topAppBarText = stringResource(id = R.string.sign_in),
-            onSignedInAsGuest = { onNavigationEvent(SignInEvent.SignInAsGuest) },
-            onBackPressed = { onNavigationEvent(SignInEvent.NavigateBack) },
-            modifier = Modifier.fillMaxWidth()
-        ) {
-            Column(modifier = Modifier.fillMaxWidth()) {
-                SignInContent(
-                    onSignInSubmitted = { email, password ->
-                        onNavigationEvent(SignInEvent.SignIn(email, password))
+
+    val snackbarHostState = remember { SnackbarHostState() }
+    val scope = rememberCoroutineScope()
+
+    val snackbarErrorText = stringResource(id = R.string.feature_not_available)
+    val snackbarActionLabel = stringResource(id = R.string.dismiss)
+
+    Scaffold(
+        topBar = {
+            SignInSignUpTopAppBar(
+                topAppBarText = stringResource(id = R.string.sign_in),
+                onBackPressed = { onNavigationEvent(SignInEvent.NavigateBack) }
+            )
+        },
+        bodyContent = {
+            SignInSignUpScreen(
+                onSignedInAsGuest = { onNavigationEvent(SignInEvent.SignInAsGuest) },
+                modifier = Modifier.fillMaxWidth()
+            ) {
+                Column(modifier = Modifier.fillMaxWidth()) {
+                    SignInContent(
+                        onSignInSubmitted = { email, password ->
+                            onNavigationEvent(SignInEvent.SignIn(email, password))
+                        }
+                    )
+                    Spacer(modifier = Modifier.preferredHeight(16.dp))
+                    TextButton(
+                        onClick = {
+                            scope.launch {
+                                snackbarHostState.showSnackbar(
+                                    message = snackbarErrorText,
+                                    actionLabel = snackbarActionLabel
+                                )
+                            }
+                        },
+                        modifier = Modifier.fillMaxWidth()
+                    ) {
+                        Text(text = stringResource(id = R.string.forgot_password))
                     }
-                )
-                Spacer(modifier = Modifier.preferredHeight(16.dp))
-                TextButton(
-                    onClick = { showSnackbar.value = true },
-                    modifier = Modifier.fillMaxWidth()
-                ) {
-                    Text(text = stringResource(id = R.string.forgot_password))
                 }
             }
         }
+    )
+
+    Stack(modifier = Modifier.fillMaxSize()) {
```

Comment:
Just checking, should this be at the bottom of the screen?

Reply:
Yes. good catch.